[BUG] Response validation fails when using authentication and custom session

### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Validating responses fails when an endpoint specifies a `security` parameter and a custom test client is passed as the `session` argument to `case.call_and_validate()`, such as in [this example from the documentation](https://schemathesis.readthedocs.io/en/stable/auth.html#custom-test-client-in-python-tests).

### To Reproduce

🚨 **Mandatory** 🚨: Steps to reproduce the behavior:

Running this test example with `pytest` should reproduce the issue:

```python
from typing import Annotated
from fastapi import FastAPI, Security
from fastapi.security import APIKeyHeader
from starlette_testclient import TestClient
import schemathesis


app = FastAPI()


@app.get("/", responses={200: {"model": {}}, 403: {"model": {}}})
def root(api_key: Annotated[str, Security(APIKeyHeader(name="x-api-key"))]):
    return {"message": "Hello world"}


schemathesis.experimental.OPEN_API_3_1.enable()

schema = schemathesis.from_asgi("/openapi.json", app)


@schema.parametrize()
def test_api(case):
    client = TestClient(app)
    case.call_and_validate(session=client)
```

This gives me the following error:

```
FAILED tests/test_schema.py::test_api[GET /] - requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7c2e41ffa6...
```

This is the OpenAPI schema:

```json
{
  "openapi": "3.1.0",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "paths": {
    "/": {
      "get": {
        "summary": "Root",
        "operationId": "root__get",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {}
              }
            }
          },
          "403": {
            "description": "Forbidden"
          }
        },
        "security": [
          {
            "APIKeyHeader": []
          }
        ]
      }
    }
  },
  "components": {
    "securitySchemes": {
      "APIKeyHeader": {
        "type": "apiKey",
        "in": "header",
        "name": "x-api-key"
      }
    }
  }
}
```

### Expected behavior

The test given above should pass

### Environment

```
- OS: Linux
- Python version: 3.12
- Schemathesis version: 3.34.1
- Spec version: 3.1.0
```

### Additional context

I believe the issue is in this check: https://github.com/schemathesis/schemathesis/blob/master/src/schemathesis/specs/openapi/checks.py#L351 The check is creating a `requests.Session` which will lead to an actual HTTP call. It should be using the specified session instead.

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -4,6 +4,10 @@ Changelog
 :version:`Unreleased <v3.34.1...HEAD>` - TBD
 --------------------------------------------
 
+**Fixed**
+
+- Not using the proper session in the ``ignored_auth`` check. :issue:`2409`
+
 .. _v3.34.1:
 
 :version:`3.34.1 <v3.34.0...v3.34.1>` - 2024-08-20

--- a/src/schemathesis/specs/openapi/checks.py
+++ b/src/schemathesis/specs/openapi/checks.py
@@ -348,7 +348,8 @@ def ignored_auth(response: GenericResponse, case: Case) -> bool | None:
             # If there is auth in the request, then drop it and retry the call
             request = _remove_auth_from_request(response.request, security_parameters)
             response.request = request
-            new_response = Session().send(request)
+            session = getattr(response, "_session", None) or Session()
+            new_response = session.send(request)
             if new_response.ok:
                 # Mutate the response object in place on the best effort basis
                 for attribute in new_response.__attrs__:

--- a/src/schemathesis/transports/__init__.py
+++ b/src/schemathesis/transports/__init__.py
@@ -192,6 +192,7 @@ class RequestsTransport:
                 context=failures.RequestTimeout(message=message, timeout=timeout),
             ) from None
         response.verify = verify  # type: ignore[attr-defined]
+        response._session = session  # type: ignore[attr-defined]
         if close_session:
             session.close()
         return response

--- a/test/specs/openapi/test_ignored_auth.py
+++ b/test/specs/openapi/test_ignored_auth.py
@@ -1,9 +1,13 @@
 import json
+import sys
 from base64 import b64decode
 
 import pytest
 import requests
-from hypothesis import settings
+from fastapi import FastAPI, Security
+from fastapi.security import APIKeyHeader
+from hypothesis import given, settings
+from starlette_testclient import TestClient
 
 import schemathesis
 from schemathesis.generation import GenerationConfig
@@ -107,3 +111,26 @@ def test_remove_auth_from_case(schema_url, key, parameters):
     case = schema["/success"]["GET"].make_case(**{key: {"A": "V"}})
     _remove_auth_from_case(case, parameters)
     assert not getattr(case, key)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason="typing.Annotated is not available in Python 3.8")
+def test_proper_session():
+    from typing import Annotated
+
+    app = FastAPI()
+
+    @app.get("/", responses={200: {"model": {}}, 403: {"model": {}}})
+    def root(api_key: Annotated[str, Security(APIKeyHeader(name="x-api-key"))]):
+        return {"message": "Hello world"}
+
+    schemathesis.experimental.OPEN_API_3_1.enable()
+
+    schema = schemathesis.from_asgi("/openapi.json", app)
+
+    @given(case=schema["/"]["GET"].as_strategy())
+    @settings(max_examples=10)
+    def test(case):
+        client = TestClient(app)
+        case.call_and_validate(session=client)
+
+    test()

